usaspending_api/disaster/v2/views/recipient/count.py

```diff
@@ -0,0 +1,39 @@
+from django.db.models import F, Count
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.disaster.v2.views.count_base import CountBase
+from usaspending_api.awards.models import FinancialAccountsByAwards, TransactionFABS, TransactionFPDS
+from usaspending_api.disaster.v2.views.disaster_base import FabaOutlayMixin, AwardTypeMixin
+
+
+class RecipientCountViewSet(CountBase, FabaOutlayMixin, AwardTypeMixin):
+    """
+    Obtain the count of Recipients related to supplied DEFC filter.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/disaster/recipient/count.md"
+
+    required_filters = ["def_codes", "award_type_codes"]
+
+    @cache_response()
+    def post(self, request: Request) -> Response:
+
+        filters = [
+            self.is_in_provided_def_codes(),
+            self.all_closed_defc_submissions,
+            self.is_non_zero_award_cpe(),
+        ]
+        award_ids = FinancialAccountsByAwards.objects.filter(*filters).values("award")
+        fabs_count = TransactionFABS.objects.annotate(award=F("transaction__award__id")).filter(award__in=award_ids)
```

Comment:
What is the performance like for this in sandbox and dev? This strategy seems inefficient with the two separate calls but it could be faster to run by making two smaller queries.
Did you experiment with either of these ideas?
- two transaction SQL unioned together
- Following the ORM joins and use a coalesce

Reply:
Dev was at about 11 seconds, still working on configuration to Sandbox. I skipped too much performance optimization since it was already within parameters.